cli/src/cmd/forge/script/broadcast.rs

```diff
@@ -38,31 +43,63 @@ impl ScriptArgs {
             // otherwise.
             let sequential_broadcast = local_wallets.len() != 1 || self.slow;
 
-            let transactions = deployment_sequence.transactions.clone();
+            // Make a one-time gas price estimation
+            let gas_price = provider.get_gas_price().await?;
+            let eip1559_fees = provider.estimate_eip1559_fees(None).await?;
 
             // Iterate through transactions, matching the `from` field with the associated
             // wallet. Then send the transaction. Panics if we find a unknown `from`
-            let sequence = transactions.into_iter().map(|tx| {
-                let from = *tx.from().expect("No sender for onchain transaction!");
-                let signer = local_wallets.get(&from).expect("`find_all` returned incomplete.");
-                (tx, signer)
-            });
+            let sequence = deployment_sequence
+                .transactions
+                .iter()
+                .skip(already_broadcasted)
+                .map(|tx| {
+                    let from = *tx.from().expect("No sender for onchain transaction!");
+                    let signer = local_wallets.get(&from).expect("`find_all` returned incomplete.");
+                    let mut tx = tx.clone();
```

Comment:
what about instead of cloning doing `.iter_mut`? @mattsse should be fine?

Reply:
defer to @joshieDo 

since the `deployment_sequence` is borrowed mut, would that makes sense to set gas for the tx in place instead of cloning? however the `sequence` is then batched and executed, so looks like cloning is intended and required here

Reply:
> however the sequence is then batched and executed, so looks like cloning is intended and required here
yeah, and it needs to borrow it as mut  to `.save()` the existing receipts + timestamp